3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/utils/ConversionUtils.scala

```diff
@@ -311,52 +311,44 @@ object ConversionUtils {
     URLEncoder.encode(str, enc)
 
   /**
-   * A wrapper around Java's
-   * URI.create().
-   *
-   * Exceptions thrown by
-   * URI.create():
-   * 1. NullPointerException
-   *    if uri is null
-   * 2. IllegalArgumentException
-   *    if uri violates RFC 2396
-   *
-   * @param uri The URI string to
-   *        convert
-   * @param useNetaporter Whether to use the
-   *        com.netaporter.uri library
-   * @return an Option-boxed URI object, or an
-   *         error message, all
-   *         wrapped in a Validation
+   * Creates a [[URI]] from a string.
+   *
+   * A "relaxed" parsing can be done if `relaxed` is set to true.
+   *
+   * Few examples of URIs that would fail non-relaxed parsing:
+   * - URI having at least one % not followed by 2 hexadecimal digits
+   * - URI with more than one #
+   * - URI containing a |
+   * Such URIs would need a relaxed parsing.
+   *
+   * @param uri String containing the URI to parse.
+   * @param relaxed Specifies whether parsing should be relaxed or not.
+   *
+   * @return [[Validation]] wrapping the result of the parsing:
+   *         - [[Success]] with the parsed URI if there was no error or with [[None]] if the input was `null`.
+   *         - [[Failure]] with the error message if something went wrong.
    */
-  def stringToUri(uri: String, useNetaporter: Boolean = false): Validation[String, Option[URI]] =
+  def stringToUri(uri: String, relaxed: Boolean = false): Validation[String, Option[URI]] =
     try {
-      val r = uri.replaceAll(" ", "%20") // Because so many raw URIs are bad, #346
-      Some(URI.create(r)).success
+      Option(uri).map(_.replaceAll(" ", "%20")).map(URI.create).success
     } catch {
-      case npe: NullPointerException => None.success
-      case iae: IllegalArgumentException =>
-        useNetaporter match {
-          case false => {
-            val netaporterUri = try {
-              Uri.parse(uri).success
+      case e: Exception =>
+        relaxed match {
+          case false =>
+            "Provided URI [%s] could not be parsed (without relaxed parsing). Error: [%s]"
+              .format(uri, e.getMessage)
+              .fail
+          case true =>
+            try {
+              implicit val config = UriConfig(decoder = NoopDecoder) // no percent-decoding
+              Option(uri).map(Uri.parse).map(_.toJavaURI).success
```

Comment:
there is https://github.com/lemonlabsuk/scala-uri/blob/07756b8e2eb7de1318767edbc730cee6b768b736/shared/src/main/scala/io/lemonlabs/uri/Uri.scala#L556-L560